src/app/teams/teams-view.component.ts

```diff
@@ -192,7 +192,11 @@ export class TeamsViewComponent implements OnInit, OnDestroy {
             this.planetMessageService.showMessage(`You have ${config.successMsg} ${displayName}`);
             this.team = change === 'course' ? res : this.team;
             if (res.status === 'archived') {
-              this.router.navigate([ '/teams' ]);
+              if (this.mode === 'team') {
```

Comment:
There is a `goBack` method which takes care of this logic.  Please use that so it makes it more maintainable (one place to change if we need to add logic for the routing).